v8isolate.cc

```diff
@@ -14,14 +14,31 @@ void* ArrayBufferAllocator::AllocateUninitialized(size_t length) {
 
 void ArrayBufferAllocator::Free(void* data, size_t) { free(data); }
 
-V8Isolate::V8Isolate() {
+V8Isolate::V8Isolate() : startup_data{} {
   v8::Isolate::CreateParams create_params;
   create_params.array_buffer_allocator = &allocator;
   isolate_ = v8::Isolate::New(create_params);
 }
 
+V8Isolate::V8Isolate(const char* snapshot_js_str) {
+  startup_data = v8::V8::CreateSnapshotDataBlob(snapshot_js_str);
+  if (startup_data.data == NULL) {
+    throw BadSnapshotDataException();
```

Comment:
> Doesn't this introduce an issue if the snapshot blob becomes unreachable in go and is
> garbage collected/freed, but the isolate still references it on the C++ side? I'd vote for tying > the snapshot and isolate lifetimes together to make things simpler here.

Sure enough, I think that's a bug in my code.

> I'm seeing CreateSnapshotDataBlob return a NULL data pointer when it is passed bad JS. Are there situations where it would return a valid data pointer despite receiving bad JS?

Again, I think you're exactly right.  Thanks, I'll update my code here too.